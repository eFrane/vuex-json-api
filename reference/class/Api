<!DOCTYPE html>
<html lang="en-US">
  <head>
    <meta charset="utf-8">
    <meta name="viewport" content="width=device-width,initial-scale=1">
    <title>Json:Api for Vuex</title>
    <meta name="generator" content="VuePress 1.8.0">
    
    <meta name="description" content="Automagic Json:Api integration for Vuex">
    
    <link rel="preload" href="/assets/css/0.styles.ba26daf8.css" as="style"><link rel="preload" href="/assets/js/app.bbfb528b.js" as="script"><link rel="preload" href="/assets/js/2.0dafffee.js" as="script"><link rel="preload" href="/assets/js/25.4f939962.js" as="script"><link rel="prefetch" href="/assets/js/10.6f689e45.js"><link rel="prefetch" href="/assets/js/11.3e854a07.js"><link rel="prefetch" href="/assets/js/12.5e6fc63d.js"><link rel="prefetch" href="/assets/js/13.b6b5cdaa.js"><link rel="prefetch" href="/assets/js/14.c2149562.js"><link rel="prefetch" href="/assets/js/15.3b44d0e0.js"><link rel="prefetch" href="/assets/js/16.5963f83e.js"><link rel="prefetch" href="/assets/js/17.e10ac304.js"><link rel="prefetch" href="/assets/js/18.557daba9.js"><link rel="prefetch" href="/assets/js/19.d48ac0dd.js"><link rel="prefetch" href="/assets/js/20.f830c59d.js"><link rel="prefetch" href="/assets/js/21.47a4ed11.js"><link rel="prefetch" href="/assets/js/22.cb5d3132.js"><link rel="prefetch" href="/assets/js/23.7c076f99.js"><link rel="prefetch" href="/assets/js/24.6ac270b7.js"><link rel="prefetch" href="/assets/js/26.0bf0a7fd.js"><link rel="prefetch" href="/assets/js/27.99d4a990.js"><link rel="prefetch" href="/assets/js/28.cdd1dd4b.js"><link rel="prefetch" href="/assets/js/29.4e9281f4.js"><link rel="prefetch" href="/assets/js/3.63095073.js"><link rel="prefetch" href="/assets/js/30.fbd48ebc.js"><link rel="prefetch" href="/assets/js/31.67d4c9e9.js"><link rel="prefetch" href="/assets/js/32.aa9038ac.js"><link rel="prefetch" href="/assets/js/33.228977a9.js"><link rel="prefetch" href="/assets/js/34.8be564c9.js"><link rel="prefetch" href="/assets/js/35.6b3da8ae.js"><link rel="prefetch" href="/assets/js/36.329aa466.js"><link rel="prefetch" href="/assets/js/37.00eac06c.js"><link rel="prefetch" href="/assets/js/38.aed504b0.js"><link rel="prefetch" href="/assets/js/39.1223306c.js"><link rel="prefetch" href="/assets/js/4.bfe88385.js"><link rel="prefetch" href="/assets/js/40.04386673.js"><link rel="prefetch" href="/assets/js/41.15514872.js"><link rel="prefetch" href="/assets/js/42.ac8985e6.js"><link rel="prefetch" href="/assets/js/43.75e46776.js"><link rel="prefetch" href="/assets/js/44.f73cfb01.js"><link rel="prefetch" href="/assets/js/45.e2c42640.js"><link rel="prefetch" href="/assets/js/46.8c55b03e.js"><link rel="prefetch" href="/assets/js/47.f35fbef8.js"><link rel="prefetch" href="/assets/js/48.8b580e70.js"><link rel="prefetch" href="/assets/js/49.57c4a9c9.js"><link rel="prefetch" href="/assets/js/5.cb249b63.js"><link rel="prefetch" href="/assets/js/50.3fd1e7cb.js"><link rel="prefetch" href="/assets/js/51.4c6b4362.js"><link rel="prefetch" href="/assets/js/52.264f1842.js"><link rel="prefetch" href="/assets/js/53.49f54f43.js"><link rel="prefetch" href="/assets/js/54.4a3a9634.js"><link rel="prefetch" href="/assets/js/55.933fb422.js"><link rel="prefetch" href="/assets/js/56.f2e73cb2.js"><link rel="prefetch" href="/assets/js/57.b9231f84.js"><link rel="prefetch" href="/assets/js/58.83ee0600.js"><link rel="prefetch" href="/assets/js/59.adde428b.js"><link rel="prefetch" href="/assets/js/6.4155846f.js"><link rel="prefetch" href="/assets/js/60.e3dfd946.js"><link rel="prefetch" href="/assets/js/61.f44f8aa3.js"><link rel="prefetch" href="/assets/js/62.1a8322f7.js"><link rel="prefetch" href="/assets/js/63.41a5a998.js"><link rel="prefetch" href="/assets/js/64.7c1cb518.js"><link rel="prefetch" href="/assets/js/65.b037b74d.js"><link rel="prefetch" href="/assets/js/66.a4187ea3.js"><link rel="prefetch" href="/assets/js/67.95fb4f5e.js"><link rel="prefetch" href="/assets/js/7.6ed4aaa9.js"><link rel="prefetch" href="/assets/js/8.58c932af.js"><link rel="prefetch" href="/assets/js/9.9a6f2cf9.js">
    <link rel="stylesheet" href="/assets/css/0.styles.ba26daf8.css">
  </head>
  <body>
    <div id="app" data-server-rendered="true"><div class="theme-container no-sidebar"><header class="navbar"><div class="sidebar-button"><svg xmlns="http://www.w3.org/2000/svg" aria-hidden="true" role="img" viewBox="0 0 448 512" class="icon"><path fill="currentColor" d="M436 124H12c-6.627 0-12-5.373-12-12V80c0-6.627 5.373-12 12-12h424c6.627 0 12 5.373 12 12v32c0 6.627-5.373 12-12 12zm0 160H12c-6.627 0-12-5.373-12-12v-32c0-6.627 5.373-12 12-12h424c6.627 0 12 5.373 12 12v32c0 6.627-5.373 12-12 12zm0 160H12c-6.627 0-12-5.373-12-12v-32c0-6.627 5.373-12 12-12h424c6.627 0 12 5.373 12 12v32c0 6.627-5.373 12-12 12z"></path></svg></div> <a href="/" class="home-link router-link-active"><!----> <span class="site-name">Json:Api for Vuex</span></a> <div class="links"><div class="search-box"><input aria-label="Search" autocomplete="off" spellcheck="false" value=""> <!----></div> <nav class="nav-links can-hide"><div class="nav-item"><a href="/" class="nav-link">
  Home
</a></div><div class="nav-item"><a href="/reference/" class="nav-link router-link-active">
  Reference
</a></div><div class="nav-item"><a href="/usage/" class="nav-link">
  Usage
</a></div> <a href="https://github.com/eFrane/vuex-json-api" target="_blank" rel="noopener noreferrer" class="repo-link">
    GitHub
    <span><svg xmlns="http://www.w3.org/2000/svg" aria-hidden="true" focusable="false" x="0px" y="0px" viewBox="0 0 100 100" width="15" height="15" class="icon outbound"><path fill="currentColor" d="M18.8,85.1h56l0,0c2.2,0,4-1.8,4-4v-32h-8v28h-48v-48h28v-8h-32l0,0c-2.2,0-4,1.8-4,4v56C14.8,83.3,16.6,85.1,18.8,85.1z"></path> <polygon fill="currentColor" points="45.7,48.7 51.3,54.3 77.2,28.5 77.2,37.2 85.2,37.2 85.2,14.9 62.8,14.9 62.8,22.9 71.5,22.9"></polygon></svg> <span class="sr-only">(opens new window)</span></span></a></nav></div></header> <div class="sidebar-mask"></div> <aside class="sidebar"><nav class="nav-links"><div class="nav-item"><a href="/" class="nav-link">
  Home
</a></div><div class="nav-item"><a href="/reference/" class="nav-link router-link-active">
  Reference
</a></div><div class="nav-item"><a href="/usage/" class="nav-link">
  Usage
</a></div> <a href="https://github.com/eFrane/vuex-json-api" target="_blank" rel="noopener noreferrer" class="repo-link">
    GitHub
    <span><svg xmlns="http://www.w3.org/2000/svg" aria-hidden="true" focusable="false" x="0px" y="0px" viewBox="0 0 100 100" width="15" height="15" class="icon outbound"><path fill="currentColor" d="M18.8,85.1h56l0,0c2.2,0,4-1.8,4-4v-32h-8v28h-48v-48h28v-8h-32l0,0c-2.2,0-4,1.8-4,4v56C14.8,83.3,16.6,85.1,18.8,85.1z"></path> <polygon fill="currentColor" points="45.7,48.7 51.3,54.3 77.2,28.5 77.2,37.2 85.2,37.2 85.2,14.9 62.8,14.9 62.8,22.9 71.5,22.9"></polygon></svg> <span class="sr-only">(opens new window)</span></span></a></nav>  <!----> </aside> <main class="page"> <div class="theme-default-content content__default"><p><a name="Api"></a></p> <h2 id="api"><a href="#api" class="header-anchor">#</a> Api</h2> <p><strong>Kind</strong>: global class</p> <ul><li><a href="#Api">Api</a> <ul><li><a href="#new_Api_new">new Api()</a></li> <li><a href="#Api+setPreprocessingCallbacks">.setPreprocessingCallbacks(callbacks)</a></li> <li><a href="#Api+setErrorCallbacks">.setErrorCallbacks(callbacks)</a></li> <li><a href="#Api+addPreprocessingCallback">.addPreprocessingCallback(callback)</a></li> <li><a href="#Api+addErrorCallback">.addErrorCallback(callback)</a></li> <li><a href="#Api+resetPreprocessing">.resetPreprocessing()</a></li> <li><a href="#Api+resetErrorCallbacks">.resetErrorCallbacks()</a></li> <li><a href="#Api+addHeaders">.addHeaders(headers)</a></li></ul></li></ul> <p><a name="new_Api_new"></a></p> <h3 id="new-api"><a href="#new-api" class="header-anchor">#</a> new Api()</h3> <p>Wrapper around json:api requests, sets content type and other defaults.</p> <p><a name="Api+setPreprocessingCallbacks"></a></p> <h3 id="api-setpreprocessingcallbacks-callbacks"><a href="#api-setpreprocessingcallbacks-callbacks" class="header-anchor">#</a> api.setPreprocessingCallbacks(callbacks)</h3> <p><strong>Kind</strong>: instance method of <a href="#Api"><code>Api</code></a></p> <table><thead><tr><th>Param</th> <th>Type</th></tr></thead> <tbody><tr><td>callbacks</td> <td><code>Array</code></td></tr></tbody></table> <p><a name="Api+setErrorCallbacks"></a></p> <h3 id="api-seterrorcallbacks-callbacks"><a href="#api-seterrorcallbacks-callbacks" class="header-anchor">#</a> api.setErrorCallbacks(callbacks)</h3> <p><strong>Kind</strong>: instance method of <a href="#Api"><code>Api</code></a></p> <table><thead><tr><th>Param</th> <th>Type</th></tr></thead> <tbody><tr><td>callbacks</td> <td><code>Array</code></td></tr></tbody></table> <p><a name="Api+addPreprocessingCallback"></a></p> <h3 id="api-addpreprocessingcallback-callback"><a href="#api-addpreprocessingcallback-callback" class="header-anchor">#</a> api.addPreprocessingCallback(callback)</h3> <p><strong>Kind</strong>: instance method of <a href="#Api"><code>Api</code></a></p> <table><thead><tr><th>Param</th> <th>Type</th></tr></thead> <tbody><tr><td>callback</td> <td><code>function</code></td></tr></tbody></table> <p><a name="Api+addErrorCallback"></a></p> <h3 id="api-adderrorcallback-callback"><a href="#api-adderrorcallback-callback" class="header-anchor">#</a> api.addErrorCallback(callback)</h3> <p><strong>Kind</strong>: instance method of <a href="#Api"><code>Api</code></a></p> <table><thead><tr><th>Param</th> <th>Type</th></tr></thead> <tbody><tr><td>callback</td> <td><code>function</code></td></tr></tbody></table> <p><a name="Api+resetPreprocessing"></a></p> <h3 id="api-resetpreprocessing"><a href="#api-resetpreprocessing" class="header-anchor">#</a> api.resetPreprocessing()</h3> <p>Reset the response preprocessing to the default behaviour</p> <p><strong>Kind</strong>: instance method of <a href="#Api"><code>Api</code></a><br> <a name="Api+resetErrorCallbacks"></a></p> <h3 id="api-reseterrorcallbacks"><a href="#api-reseterrorcallbacks" class="header-anchor">#</a> api.resetErrorCallbacks()</h3> <p>Reset the response error to the default behaviour</p> <p><strong>Kind</strong>: instance method of <a href="#Api"><code>Api</code></a><br> <a name="Api+addHeaders"></a></p> <h3 id="api-addheaders-headers"><a href="#api-addheaders-headers" class="header-anchor">#</a> api.addHeaders(headers)</h3> <p><strong>Kind</strong>: instance method of <a href="#Api"><code>Api</code></a></p> <table><thead><tr><th>Param</th> <th>Type</th></tr></thead> <tbody><tr><td>headers</td> <td><code>Object</code></td></tr></tbody></table></div> <footer class="page-edit"><!----> <!----></footer> <div class="page-nav"><p class="inner"><span class="prev">
      ←
      <a href="/reference/" class="prev router-link-active">
        Reference (JS)
      </a></span> <!----></p></div> </main></div><div class="global-ui"></div></div>
    <script src="/assets/js/app.bbfb528b.js" defer></script><script src="/assets/js/2.0dafffee.js" defer></script><script src="/assets/js/25.4f939962.js" defer></script>
  </body>
</html>
