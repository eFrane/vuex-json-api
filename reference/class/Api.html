<!DOCTYPE html>
<html lang="en-US">
  <head>
    <meta charset="utf-8">
    <meta name="viewport" content="width=device-width,initial-scale=1">
    <title>Json:Api for Vuex</title>
    <meta name="generator" content="VuePress 1.9.10">
    
    <meta name="description" content="Automagic Json:Api integration for Vuex">
    
    <link rel="preload" href="/assets/css/0.styles.769add7b.css" as="style"><link rel="preload" href="/assets/js/app.a8f03850.js" as="script"><link rel="preload" href="/assets/js/2.fb51603b.js" as="script"><link rel="preload" href="/assets/js/1.2ea433ec.js" as="script"><link rel="preload" href="/assets/js/30.1567c6aa.js" as="script"><link rel="prefetch" href="/assets/js/10.5f385d4d.js"><link rel="prefetch" href="/assets/js/11.ba710dea.js"><link rel="prefetch" href="/assets/js/12.46048386.js"><link rel="prefetch" href="/assets/js/13.357634dc.js"><link rel="prefetch" href="/assets/js/14.65769da9.js"><link rel="prefetch" href="/assets/js/15.b2acdef1.js"><link rel="prefetch" href="/assets/js/16.81bd89c9.js"><link rel="prefetch" href="/assets/js/17.459cf0b5.js"><link rel="prefetch" href="/assets/js/18.285b93d5.js"><link rel="prefetch" href="/assets/js/19.2b3d35ee.js"><link rel="prefetch" href="/assets/js/20.b6603841.js"><link rel="prefetch" href="/assets/js/21.ea8b26f4.js"><link rel="prefetch" href="/assets/js/22.dcb39a39.js"><link rel="prefetch" href="/assets/js/23.3a6a38e2.js"><link rel="prefetch" href="/assets/js/24.3226531c.js"><link rel="prefetch" href="/assets/js/25.bf01f32a.js"><link rel="prefetch" href="/assets/js/26.62a29a7a.js"><link rel="prefetch" href="/assets/js/27.87f07c1c.js"><link rel="prefetch" href="/assets/js/28.95b71fce.js"><link rel="prefetch" href="/assets/js/29.2626e755.js"><link rel="prefetch" href="/assets/js/3.bee1657c.js"><link rel="prefetch" href="/assets/js/31.2b529aa9.js"><link rel="prefetch" href="/assets/js/32.afee7aa5.js"><link rel="prefetch" href="/assets/js/33.83231f93.js"><link rel="prefetch" href="/assets/js/34.fba2356b.js"><link rel="prefetch" href="/assets/js/35.1e223504.js"><link rel="prefetch" href="/assets/js/36.f6a75ce8.js"><link rel="prefetch" href="/assets/js/37.c42e0dd6.js"><link rel="prefetch" href="/assets/js/38.ba646047.js"><link rel="prefetch" href="/assets/js/39.4f6dc3a8.js"><link rel="prefetch" href="/assets/js/4.52dc2f05.js"><link rel="prefetch" href="/assets/js/40.9cf469f9.js"><link rel="prefetch" href="/assets/js/41.e2982e59.js"><link rel="prefetch" href="/assets/js/42.0e52e9a2.js"><link rel="prefetch" href="/assets/js/43.2250b9a9.js"><link rel="prefetch" href="/assets/js/44.511c4928.js"><link rel="prefetch" href="/assets/js/45.586c49c5.js"><link rel="prefetch" href="/assets/js/46.ecd1fc23.js"><link rel="prefetch" href="/assets/js/47.1554335c.js"><link rel="prefetch" href="/assets/js/48.77cb5f20.js"><link rel="prefetch" href="/assets/js/49.b7c951c8.js"><link rel="prefetch" href="/assets/js/5.18620486.js"><link rel="prefetch" href="/assets/js/50.6beb7ee7.js"><link rel="prefetch" href="/assets/js/51.3db5f412.js"><link rel="prefetch" href="/assets/js/52.3fae7d23.js"><link rel="prefetch" href="/assets/js/53.0a5a39dd.js"><link rel="prefetch" href="/assets/js/54.7afc0b6c.js"><link rel="prefetch" href="/assets/js/55.9bb55334.js"><link rel="prefetch" href="/assets/js/56.907e5f15.js"><link rel="prefetch" href="/assets/js/57.a2e0b921.js"><link rel="prefetch" href="/assets/js/58.edbfbb04.js"><link rel="prefetch" href="/assets/js/59.c26a30a4.js"><link rel="prefetch" href="/assets/js/6.891bb6ed.js"><link rel="prefetch" href="/assets/js/60.9765273a.js"><link rel="prefetch" href="/assets/js/61.e1898988.js"><link rel="prefetch" href="/assets/js/62.45996bc1.js"><link rel="prefetch" href="/assets/js/63.ce848b16.js"><link rel="prefetch" href="/assets/js/64.2790a4b5.js"><link rel="prefetch" href="/assets/js/65.052dd7c1.js"><link rel="prefetch" href="/assets/js/66.9fee64b0.js"><link rel="prefetch" href="/assets/js/67.007a4b9c.js"><link rel="prefetch" href="/assets/js/68.30347128.js"><link rel="prefetch" href="/assets/js/69.751a5ab9.js"><link rel="prefetch" href="/assets/js/7.243f6fed.js"><link rel="prefetch" href="/assets/js/70.ed0081f6.js"><link rel="prefetch" href="/assets/js/71.568c0c6b.js"><link rel="prefetch" href="/assets/js/72.0b11e488.js"><link rel="prefetch" href="/assets/js/73.6a64cd6d.js"><link rel="prefetch" href="/assets/js/74.6da308f8.js"><link rel="prefetch" href="/assets/js/vendors~docsearch.cdc87792.js">
    <link rel="stylesheet" href="/assets/css/0.styles.769add7b.css">
  </head>
  <body>
    <div id="app" data-server-rendered="true"><div class="theme-container"><header class="navbar"><div class="sidebar-button"><svg xmlns="http://www.w3.org/2000/svg" aria-hidden="true" role="img" viewBox="0 0 448 512" class="icon"><path fill="currentColor" d="M436 124H12c-6.627 0-12-5.373-12-12V80c0-6.627 5.373-12 12-12h424c6.627 0 12 5.373 12 12v32c0 6.627-5.373 12-12 12zm0 160H12c-6.627 0-12-5.373-12-12v-32c0-6.627 5.373-12 12-12h424c6.627 0 12 5.373 12 12v32c0 6.627-5.373 12-12 12zm0 160H12c-6.627 0-12-5.373-12-12v-32c0-6.627 5.373-12 12-12h424c6.627 0 12 5.373 12 12v32c0 6.627-5.373 12-12 12z"></path></svg></div> <a href="/" class="home-link router-link-active"><!----> <span class="site-name">Json:Api for Vuex</span></a> <div class="links"><div class="search-box"><input aria-label="Search" autocomplete="off" spellcheck="false" value=""> <!----></div> <nav class="nav-links can-hide"><div class="nav-item"><a href="/" class="nav-link">
  Home
</a></div><div class="nav-item"><a href="/development/" class="nav-link">
  Development
</a></div><div class="nav-item"><a href="/reference/" class="nav-link router-link-active">
  Reference
</a></div><div class="nav-item"><a href="/usage/" class="nav-link">
  Usage
</a></div> <a href="https://github.com/eFrane/vuex-json-api" target="_blank" rel="noopener noreferrer" class="repo-link">
    GitHub
    <span><svg xmlns="http://www.w3.org/2000/svg" aria-hidden="true" focusable="false" x="0px" y="0px" viewBox="0 0 100 100" width="15" height="15" class="icon outbound"><path fill="currentColor" d="M18.8,85.1h56l0,0c2.2,0,4-1.8,4-4v-32h-8v28h-48v-48h28v-8h-32l0,0c-2.2,0-4,1.8-4,4v56C14.8,83.3,16.6,85.1,18.8,85.1z"></path> <polygon fill="currentColor" points="45.7,48.7 51.3,54.3 77.2,28.5 77.2,37.2 85.2,37.2 85.2,14.9 62.8,14.9 62.8,22.9 71.5,22.9"></polygon></svg> <span class="sr-only">(opens new window)</span></span></a></nav></div></header> <div class="sidebar-mask"></div> <aside class="sidebar"><nav class="nav-links"><div class="nav-item"><a href="/" class="nav-link">
  Home
</a></div><div class="nav-item"><a href="/development/" class="nav-link">
  Development
</a></div><div class="nav-item"><a href="/reference/" class="nav-link router-link-active">
  Reference
</a></div><div class="nav-item"><a href="/usage/" class="nav-link">
  Usage
</a></div> <a href="https://github.com/eFrane/vuex-json-api" target="_blank" rel="noopener noreferrer" class="repo-link">
    GitHub
    <span><svg xmlns="http://www.w3.org/2000/svg" aria-hidden="true" focusable="false" x="0px" y="0px" viewBox="0 0 100 100" width="15" height="15" class="icon outbound"><path fill="currentColor" d="M18.8,85.1h56l0,0c2.2,0,4-1.8,4-4v-32h-8v28h-48v-48h28v-8h-32l0,0c-2.2,0-4,1.8-4,4v56C14.8,83.3,16.6,85.1,18.8,85.1z"></path> <polygon fill="currentColor" points="45.7,48.7 51.3,54.3 77.2,28.5 77.2,37.2 85.2,37.2 85.2,14.9 62.8,14.9 62.8,22.9 71.5,22.9"></polygon></svg> <span class="sr-only">(opens new window)</span></span></a></nav>  <ul class="sidebar-links"><li><section class="sidebar-group depth-0"><p class="sidebar-heading open"><span></span> <!----></p> <ul class="sidebar-links sidebar-group-items"><li><a href="/reference/class/Api.html#api" class="sidebar-link">Api</a><ul class="sidebar-sub-headers"><li class="sidebar-sub-header"><a href="/reference/class/Api.html#new-api" class="sidebar-link">new Api()</a></li><li class="sidebar-sub-header"><a href="/reference/class/Api.html#api-setsuccesscallbacks-callbacks" class="sidebar-link">api.setSuccessCallbacks(callbacks)</a></li><li class="sidebar-sub-header"><a href="/reference/class/Api.html#api-seterrorcallbacks-callbacks" class="sidebar-link">api.setErrorCallbacks(callbacks)</a></li><li class="sidebar-sub-header"><a href="/reference/class/Api.html#api-addsuccesscallback-callback" class="sidebar-link">api.addSuccessCallback(callback)</a></li><li class="sidebar-sub-header"><a href="/reference/class/Api.html#api-adderrorcallback-callback" class="sidebar-link">api.addErrorCallback(callback)</a></li><li class="sidebar-sub-header"><a href="/reference/class/Api.html#api-resetsuccesscallbacks" class="sidebar-link">api.resetSuccessCallbacks()</a></li><li class="sidebar-sub-header"><a href="/reference/class/Api.html#api-reseterrorcallbacks" class="sidebar-link">api.resetErrorCallbacks()</a></li><li class="sidebar-sub-header"><a href="/reference/class/Api.html#api-setheader-name-value-overwrite" class="sidebar-link">api.setHeader(name, value, overwrite)</a></li><li class="sidebar-sub-header"><a href="/reference/class/Api.html#api-addheaders-headers" class="sidebar-link">api.addHeaders(headers)</a></li><li class="sidebar-sub-header"><a href="/reference/class/Api.html#api-dorequest-method-url-params-data-⇒-promise-response" class="sidebar-link">api._doRequest(method, url, params, data) ⇒</a></li></ul></li></ul></section></li></ul> </aside> <main class="page"> <div class="theme-default-content content__default"><p><a name="Api"></a></p> <h2 id="api"><a href="#api" class="header-anchor">#</a> Api</h2> <p><strong>Kind</strong>: global class</p> <ul><li><a href="#Api">Api</a> <ul><li><a href="#new_Api_new">new Api()</a></li> <li><a href="#Api+setSuccessCallbacks">.setSuccessCallbacks(callbacks)</a></li> <li><a href="#Api+setErrorCallbacks">.setErrorCallbacks(callbacks)</a></li> <li><a href="#Api+addSuccessCallback">.addSuccessCallback(callback)</a></li> <li><a href="#Api+addErrorCallback">.addErrorCallback(callback)</a></li> <li><a href="#Api+resetSuccessCallbacks">.resetSuccessCallbacks()</a></li> <li><a href="#Api+resetErrorCallbacks">.resetErrorCallbacks()</a></li> <li><a href="#Api+setHeader">.setHeader(name, value, overwrite)</a></li> <li><a href="#Api+addHeaders">.addHeaders(headers)</a></li> <li><a href="#Api+_doRequest">._doRequest(method, url, params, data)</a> ⇒ <code>Promise.&lt;Response&gt;</code></li></ul></li></ul> <p><a name="new_Api_new"></a></p> <h3 id="new-api"><a href="#new-api" class="header-anchor">#</a> new Api()</h3> <p>Wrapper around json:api 1.0 requests, sets content type and other defaults.</p> <p><a name="Api+setSuccessCallbacks"></a></p> <h3 id="api-setsuccesscallbacks-callbacks"><a href="#api-setsuccesscallbacks-callbacks" class="header-anchor">#</a> api.setSuccessCallbacks(callbacks)</h3> <p>Functions to be called after successful requests.</p> <p><strong>Kind</strong>: instance method of <a href="#Api"><code>Api</code></a></p> <table><thead><tr><th>Param</th> <th>Type</th></tr></thead> <tbody><tr><td>callbacks</td> <td><code>Array.&lt;function()&gt;</code></td></tr></tbody></table> <p><a name="Api+setErrorCallbacks"></a></p> <h3 id="api-seterrorcallbacks-callbacks"><a href="#api-seterrorcallbacks-callbacks" class="header-anchor">#</a> api.setErrorCallbacks(callbacks)</h3> <p><strong>Kind</strong>: instance method of <a href="#Api"><code>Api</code></a></p> <table><thead><tr><th>Param</th> <th>Type</th></tr></thead> <tbody><tr><td>callbacks</td> <td><code>Array.&lt;function()&gt;</code></td></tr></tbody></table> <p><a name="Api+addSuccessCallback"></a></p> <h3 id="api-addsuccesscallback-callback"><a href="#api-addsuccesscallback-callback" class="header-anchor">#</a> api.addSuccessCallback(callback)</h3> <p><strong>Kind</strong>: instance method of <a href="#Api"><code>Api</code></a></p> <table><thead><tr><th>Param</th> <th>Type</th></tr></thead> <tbody><tr><td>callback</td> <td><code>function</code></td></tr></tbody></table> <p><a name="Api+addErrorCallback"></a></p> <h3 id="api-adderrorcallback-callback"><a href="#api-adderrorcallback-callback" class="header-anchor">#</a> api.addErrorCallback(callback)</h3> <p><strong>Kind</strong>: instance method of <a href="#Api"><code>Api</code></a></p> <table><thead><tr><th>Param</th> <th>Type</th></tr></thead> <tbody><tr><td>callback</td> <td><code>function</code></td></tr></tbody></table> <p><a name="Api+resetSuccessCallbacks"></a></p> <h3 id="api-resetsuccesscallbacks"><a href="#api-resetsuccesscallbacks" class="header-anchor">#</a> api.resetSuccessCallbacks()</h3> <p>Reset the response success handling to the default behaviour</p> <p><strong>Kind</strong>: instance method of <a href="#Api"><code>Api</code></a><br> <a name="Api+resetErrorCallbacks"></a></p> <h3 id="api-reseterrorcallbacks"><a href="#api-reseterrorcallbacks" class="header-anchor">#</a> api.resetErrorCallbacks()</h3> <p>Reset the response error to the default behaviour</p> <p><strong>Kind</strong>: instance method of <a href="#Api"><code>Api</code></a><br> <a name="Api+setHeader"></a></p> <h3 id="api-setheader-name-value-overwrite"><a href="#api-setheader-name-value-overwrite" class="header-anchor">#</a> api.setHeader(name, value, overwrite)</h3> <p><strong>Kind</strong>: instance method of <a href="#Api"><code>Api</code></a></p> <table><thead><tr><th>Param</th> <th>Type</th> <th>Default</th></tr></thead> <tbody><tr><td>name</td> <td><code>string</code></td> <td></td></tr> <tr><td>value</td> <td><code>string</code></td> <td></td></tr> <tr><td>overwrite</td> <td><code>boolean</code></td> <td><code>false</code></td></tr></tbody></table> <p><a name="Api+addHeaders"></a></p> <h3 id="api-addheaders-headers"><a href="#api-addheaders-headers" class="header-anchor">#</a> api.addHeaders(headers)</h3> <p>Add multiple request headers at once by passing an object:</p> <div class="language-js extra-class"><pre class="language-js"><code>api<span class="token punctuation">.</span><span class="token function">addHeaders</span><span class="token punctuation">(</span><span class="token punctuation">{</span>
  <span class="token string-property property">'X-My-Header'</span><span class="token operator">:</span> <span class="token string">'my-header-value'</span><span class="token punctuation">,</span>
  <span class="token string-property property">'E-Tag'</span><span class="token operator">:</span> <span class="token string">'12923-434-123'</span>
<span class="token punctuation">}</span><span class="token punctuation">)</span>
</code></pre></div><p><strong>Kind</strong>: instance method of <a href="#Api"><code>Api</code></a></p> <table><thead><tr><th>Param</th> <th>Type</th></tr></thead> <tbody><tr><td>headers</td> <td><code>Object</code></td></tr></tbody></table> <p><a name="Api+_doRequest"></a></p> <h3 id="api-dorequest-method-url-params-data-⇒-promise-response"><a href="#api-dorequest-method-url-params-data-⇒-promise-response" class="header-anchor">#</a> api._doRequest(method, url, params, data) ⇒ <code>Promise.&lt;Response&gt;</code></h3> <p><strong>Kind</strong>: instance method of <a href="#Api"><code>Api</code></a><br> <strong>Access</strong>: protected</p> <table><thead><tr><th>Param</th> <th>Type</th></tr></thead> <tbody><tr><td>method</td> <td><code>string</code></td></tr> <tr><td>url</td> <td><code>string</code></td></tr> <tr><td>params</td> <td><code>Object</code></td></tr> <tr><td>data</td> <td><code>Object</code></td></tr></tbody></table></div> <footer class="page-edit"><!----> <!----></footer> <div class="page-nav"><p class="inner"><span class="prev">
      ←
      <a href="/reference/" class="prev router-link-active">
        Reference (JS)
      </a></span> <!----></p></div> </main></div><div class="global-ui"></div></div>
    <script src="/assets/js/app.a8f03850.js" defer></script><script src="/assets/js/2.fb51603b.js" defer></script><script src="/assets/js/1.2ea433ec.js" defer></script><script src="/assets/js/30.1567c6aa.js" defer></script>
  </body>
</html>
