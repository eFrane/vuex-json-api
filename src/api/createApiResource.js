export function createApiResource () {
  console.time('api: add method proxies for route ' + routeName)

  for (let methodName in methods) {
    if (methods.hasOwnProperty(methodName)) {
      let route = methods[methodName]

      if (methodName.indexOf('related.') === 0) {
        this.registerRelatedResourceMethod(routeName, methodName, route)
        continue
      }

      this[routeName][methodName] = createApiResourceMethodProxy(this, methodName, route)
    }
  }

  console.timeEnd('api: add method proxies for route ' + routeName)
}
