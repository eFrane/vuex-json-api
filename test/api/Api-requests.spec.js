import { Api } from '@/api/Api'
import fetchMock from 'fetch-mock'

// This spec only tests the request related methods of Api

test('request sends correct headers', async () => {
  fetchMock.config.sendAsJson = false
  fetchMock.get('https://test/', { body: '{}' })

  const api = new Api()
  api.setBaseUrl('https://test')
  const response = await api._doRequest('get', '/')
  const receivedRequestOptions = fetchMock.lastOptions()

  expect(receivedRequestOptions.headers).toStrictEqual({
    Accept: 'application/vnd.api+json',
    'Content-Type': 'application/vnd.api+json'
  })

  expect(await response.json()).toEqual({})
})

test('verb methods pass to doRequest', () => {
  const nonRequestingApi = new Api()
  nonRequestingApi.doRequest = jest.fn()

  nonRequestingApi.get('/')
  expect(nonRequestingApi.doRequest.mock.calls[0]).toEqual(['get', '/', null, null])

  nonRequestingApi.post('/')
  expect(nonRequestingApi.doRequest.mock.calls[1]).toEqual(['post', '/', null, null])

  nonRequestingApi.put('/')
  expect(nonRequestingApi.doRequest.mock.calls[2]).toEqual(['put', '/', null, null])

  nonRequestingApi.patch('/')
  expect(nonRequestingApi.doRequest.mock.calls[3]).toEqual(['patch', '/', null, null])

  nonRequestingApi.delete('/')
  expect(nonRequestingApi.doRequest.mock.calls[4]).toEqual(['delete', '/', null, null])
})
