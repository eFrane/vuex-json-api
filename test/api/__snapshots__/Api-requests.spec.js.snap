// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`response is sealed 1`] = `"Cannot define property new_prop, object is not extensible"`;
